gfx-memory/src/allocator/linear.rs

```diff
@@ -234,40 +262,49 @@ impl<B: Backend> Allocator<B> for LinearAllocator<B> {
             }
         }
 
-        log::trace!("Allocated 'Line' of size of {}", self.line_size);
-        let (memory, ptr) = unsafe {
-            super::allocate_memory_helper(
-                device,
-                self.memory_type,
-                self.line_size,
-                self.memory_properties,
-                self.non_coherent_atom_size,
-            )?
-        };
+        let (mut line, new_allocation_size) = if let Some(line) = self.unused_lines.pop() {
+            (line, 0)
+        } else {
+            log::trace!("Allocated `Line` of size {}", self.line_size);
+            let (memory, ptr) = unsafe {
+                super::allocate_memory_helper(
+                    device,
+                    self.memory_type,
+                    self.line_size,
+                    self.memory_properties,
+                    self.non_coherent_atom_size,
+                )?
+            };
 
-        let line = Line {
-            allocated: size,
-            freed: 0,
-            ptr,
-            memory: Arc::new(memory),
+            (
+                Line {
+                    allocated: size,
+                    freed: 0,
+                    ptr,
+                    memory: Arc::new(memory),
+                },
+                self.line_size,
+            )
         };
+        line.allocated = size;
+        line.freed = 0;
 
         let block = LinearBlock {
             line_index: self.finished_lines_count + lines_count,
-            memory: Arc::clone(&line.memory),
-            ptr,
+            memory: line.memory.clone(),
```

Comment:
nit: let's consistently do `Arc::clone(&line.memory)`. This is better for being explicit. Who knows what clone() does? :)

Reply:
Ah I didnt understand why it was written like that, makes sense though.